registrations/management/commands/tests/test_teenmomconnect_postbirth_subscriptions.py

```diff
@@ -0,0 +1,377 @@
+        )
+
+    def test_validate_msisdn(self):
+        """
+        If the msisdn is valid, it should be returned in E164 international format
+        """
+        command = Command()
+        self.assertEqual(command.validate_msisdn("0820001001"), "+27820001001")
+
+    def test_validate_msisdn_not_parsable(self):
+        """
+        If it's not a parsable msisdn, an error should be logged and `None` should be
+        returned
+        """
+        command = Command()
+        command.stdout = io.StringIO()
+        self.assertEqual(command.validate_msisdn("gibberish"), None)
+        self.assertIn(
+            "Invalid phone number gibberish. Skipping...", command.stdout.getvalue()
+        )
+
+    def test_validate_msisdn_not_possible(self):
+        """
+        If it's not a possible msisdn, an error should be logged and `None` should be
+        returned
+        """
+        command = Command()
+        command.stdout = io.StringIO()
+        self.assertEqual(command.validate_msisdn("+1200012301"), None)
+        self.assertIn(
+            "Invalid phone number +1200012301. Skipping...", command.stdout.getvalue()
+        )
+
+    def test_validate_msisdn_not_valid(self):
+        """
+        If it's not a valid msisdn, an error should be logged and `None` should be
+        returned
+        """
+        command = Command()
+        command.stdout = io.StringIO()
+        self.assertEqual(command.validate_msisdn("+12001230101"), None)
+        self.assertIn(
+            "Invalid phone number +12001230101. Skipping...", command.stdout.getvalue()
+        )
+
+    def test_validate_language(self):
+        """
+        If the language code is valid, it should be returned in the Seed language format
+        """
+        command = Command()
+        self.assertEqual(command.validate_language("eng"), "eng_ZA")
+
+    def test_validate_language_invalid(self):
+        """
+        If the language code is invalid, an error should be logged and None returned
+        """
+        command = Command()
+        command.stdout = io.StringIO()
+        self.assertEqual(command.validate_language("foo"), None)
+        self.assertIn(
+            "Invalid language code foo. Skipping...", command.stdout.getvalue()
+        )
+
+    def test_validate_language_not_in_seed(self):
+        """
+        If the language code is not in the project's list of languages, an error should
+        be logged and None returned
+        """
+        command = Command()
+        command.stdout = io.StringIO()
+        self.assertEqual(command.validate_language("arg"), None)
+        self.assertIn(
+            "Invalid language code arg. Skipping...", command.stdout.getvalue()
+        )
+
+    @responses.activate
+    def test_create_or_update_identity_exists(self):
+        """
+        If the identity exists, and the language code is correct, it should be returned
+        """
+        identity_details = {
```

Comment:
I see this happens in a few places, can we move it into functions for more DRY approach?